src/xdist/workermanage.py

```diff
@@ -236,10 +238,14 @@ def shutting_down(self):
     def setup(self):
         self.log("setting up worker session")
         spec = self.gateway.spec
-        args = self.config.args
+        if hasattr(self.config, "invocation_params"):
+            args = [str(x) for x in self.config.invocation_params.args or ()]
+            option_dict = {}
+        else:
+            args = self.config.args
+            option_dict = vars(self.config.option)
         if not spec.popen or spec.chdir:
             args = make_reltoroot(self.nodemanager.roots, args)
```

Comment:
@nicoddemus I think now I correctly identified the cause of the problem. In the new code the args stripping matters cause the old option_dict is no more, while previously this stripping had no effect.

Reply:
Sorry, what "stripping" you mean? Can you please clarify?

Reply:
@nicoddemus see https://github.com/pytest-dev/pytest-xdist/pull/491 - make_reltoroot removes anything that ain't a valid path (like the arguments I need).